src/backend/sgx/ioctls.rs

```diff
@@ -187,6 +195,82 @@ impl<'a> RestrictPermissions<'a> {
     }
 }
 
+#[repr(C)]
+#[derive(Debug)]
+/// SGX_IOC_ENCLAVE_MODIFY_TYPES
+pub struct ModifyTypes<'a> {
+    /// In: starting page offset
+    offset: u64,
+    /// In: length of the address range (multiple of the page size)
+    length: u64,
+    /// In: SECINFO containing the relaxed permissions
+    secinfo: u64,
+    /// Out: ENCLU[EMODPR] return value
+    result: u64,
+    /// Out: length of the address range successfully changed
+    count: u64,
+    phantom: PhantomData<&'a ()>,
+}
+
+impl<'a> ModifyTypes<'a> {
+    /// Create a new RestrictPermissions instance.
+    pub fn new(offset: usize, length: usize, secinfo: &'a SecInfo) -> Self {
+        Self {
+            offset: offset as _,
+            length: length as _,
+            secinfo: secinfo as *const _ as _,
+            result: 0,
+            count: 0,
+            phantom: PhantomData,
+        }
+    }
+
+    /// Read result attribute.
+    pub fn result(&self) -> u64 {
+        self.count
+    }
+
+    /// Read count attribute.
+    pub fn count(&self) -> u64 {
+        self.count
+    }
```

Comment:
Why 2 different getters for the same field? Or should `result(&self)` return value of `self.result`?